src/components/App/App.js

```diff
@@ -67,13 +67,62 @@ export default class App extends React.Component {
-    // navigating directly into a specific route path from outside the app.
-    // Not sure why this is required, but it works.
+    // Setting `Router.basename` correctly is a little tricky, for two reasons:
+    //
+    // 1. Dynamic deployment to different URLs.
+    // 2. Correctly handling requests containing client-side routes.
+    //
+    // Each of these concerns has different implications, as described below.
+    //
+    // 1. Dynamic deployment to different URLs.
+    //
+    // We don't want to set `basename` to a static value. If we did that,
+    // deploying the app to a different URL would require us to modify code
+    // and rebuild the app. This rules out both a static string here and
+    // `process.env.PUBLIC_URL`, which is set by the `homepage` property in
+    // `package.json`. (The latter approach is recommended in the
+    // create-react-app documentation
+    // (https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/template/README.md#building-for-relative-paths),
+    // but it does not meet our needs here.)
+    //
+    // Instead, we set `basename` from an environment variable, `CE_BASE_PATH`,
+    // which we set at deploy time, i.e., when the app is started.
+    //
+    // 2. Correctly handling requests containing client-side routes.
+    //
+    // Requests to the application be handled in two ways:
+    //
+    // A. Externally: That is, by the server proper. This occurs when
+    //    a hyperlink to the app that originates outside the app (e.g., on
+    //    a webpage somewhere) contains a client-side route component.
+    //
+    // B. Internally: By the app iteself, which is to say, by Router. This
+    //    occurs when an internal `Link` is followed. `Router` intervenes and
+    //    no request to the server is issued.
+    //
+    // Externally handled requests can cause errors: Most servers will, without
+    // special configuration, respond to the entire request URL, not just the
+    // base path. This results in errors, since what we'd like in this case
+    // is for the server to serve the app (at the base path), not a non-existent
+    // page identified by the base path plus the client-side routing part of
+    // URL.
+    //
+    // There are two possible solutions to this problem:
+    //
+    // 1. Configure the server to ignore the client-side routing
+    // part of the URL (i.e., to care only about the base path). See the
+    // create-react-app documentation
+    //    (https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/template/README.md#serving-apps-with-client-side-routing).
+    //
+    // 2. Use a hash (#) to separate the base path from the client-side routes.
+    //    This makes the client-side route component not a part of the URL path.
+    //
+    // We choose option 2, hence the hash (#) following the base path in the
+    // `Router.basename` value below.
```

Comment:
I appreciate this detailed comment!

Has the default deployment procedure of a development docker changed at all, such as needing to manually supply a new environment variable? If so, could you edit the main readme to reflect that?